proposals/0251-font-styles.md

```diff
@@ -7,92 +7,73 @@
   .
   .
   .
-  <param name="fieldName" type="Common.TextFieldName" mandatory="true">
-    <description>The name of the field for displaying the text.</description>
-  </param>
-  <param name="fieldText" type="String" maxlength="500" mandatory="true">
+  <param name="fieldText" type="String" maxlength="1000" mandatory="true">
     <description>The  text itself.</description>
   </param>
-  <param name="fieldTypes" type="Common.MetadataType" minsize="0" maxsize="5" array="true" mandatory="false">
-    <description>The type of data contained in the field.</description>
-  </param>
-  <param name="fieldStyle" type="Common.TextStyle" mandatory="false">
-    <description>The style of text for the text field.</description>
-  </param>
-</struct>
-```
-	
-Add font styles for `PerformInteraction` `Choice` 
-
-```xml
-<struct name="Choice">
   .
   .
   .
-  <param name="textStyle" type "ChoiceTextStyle" mandatory="false">
-	<description>The style of text for each choice field.</description>
-  </param>
 </struct>
-
-<struct name="ChoiceTextStyle">
-  <param name="menuNameStyle" type="Common.TextStyle" mandatory="false">
-	<description>Text style for menuName in a choice.</description>	
-  </param>
-  <param name="secondaryTextStyle" type="Common.TextStyle" mandatory="false">
-	<description>Text style for secondaryText in a choice.</description>	
-  </param>
-  <param name="tertiaryTextStyle" type="Common.TextStyle" mandatory="false">
-	<description>Text style for tertiaryText in a choice.</description>	
-  </param>
-</struct>							  
 ```
 
+### Automatic Tag Cleanup Using Managers 
 
-## Potential downsides
+The manager API of the SDL libraries should recognize the new parameter `hyperTextSupported` and scan strings to see if unsupported elements are requested by the app. When a manager wants to generate a RPC it should scan the string and send a cleaned up version of the string in the RPC. For instance when the `ScreenManager` generates a new `Show` request it should set the main fields to a cleaned up version of the manager's text fields.
+
+Example:
+The head unit provided information that `mainField1` can support the hyper text elements `b` and `i`. If a developer sets `screenManager.textField1` to a string like `<b>Hello <u>World!</u></b>`, the manager should scan the text field and clean it up to `<b>Hello World!</b>` before setting the string to `Show.mainField1` in the RPC request.
 
-With this implementation, the whole line of text has to have the same font style. It's not possible to bold only a part of a line of text. Also this adds some HMI complexity.
+The following managers should need to scan and clean text fields before generating RPCs:
+1. ScreenManager
+2. TextAndGraphicManager
+3. SoftButtonManager
+4. ChoiceSetManager
+5. MenuManager
+
+
+## Potential downsides
+This adds some HMI complexity and would require SDL to potentially handle translating the 1000 characters in text fields to 500 characters on older headunits.
```

Comment:
Can you please specify which component of SDL would be required to handle translating the characters?  SDL Core?

Reply:
@MichaelCrimando can you please make sure this is addressed?